Tests/Source/LuaRefTests.cpp

```diff
@@ -181,38 +184,45 @@ TEST_F(LuaRefTests, Comparison)
     ASSERT_FALSE(stringA > stringAB);
     ASSERT_FALSE(stringA >= stringAB);
 
-    ASSERT_TRUE(stringA < t1);
-
-    ASSERT_TRUE(t1 == t1);
-    ASSERT_FALSE(t1 == t2);
-    ASSERT_TRUE(t1 == t3);
-    ASSERT_FALSE(t1.rawequal(t3));
-    ASSERT_FALSE(t1 == t4);
-    ASSERT_TRUE(t2 == t2);
-    ASSERT_FALSE(t2 == t3);
-    ASSERT_FALSE(t2 == t4);
-    ASSERT_TRUE(t3 == t3);
-    ASSERT_FALSE(t3 == t4);
-
-    ASSERT_FALSE(t1 < t1);
-    ASSERT_TRUE(t1 < t2);
-    ASSERT_FALSE(t1 < t3);
-    ASSERT_FALSE(t2 < t3);
-
-    ASSERT_TRUE(t1 <= t1);
-    ASSERT_TRUE(t1 <= t2);
-    ASSERT_TRUE(t1 <= t3);
-    ASSERT_FALSE(t2 <= t3);
-
-    ASSERT_FALSE(t1 > t1);
-    ASSERT_FALSE(t1 > t2);
-    ASSERT_FALSE(t1 > t3);
-    ASSERT_TRUE(t2 > t3);
-
-    ASSERT_TRUE(t1 >= t1);
-    ASSERT_FALSE(t1 >= t2);
-    ASSERT_TRUE(t1 >= t3);
-    ASSERT_TRUE(t2 >= t3);
+    ASSERT_TRUE(stringA < table1aMT);
+
+    ASSERT_TRUE(table1aMT == table1aMT);
+    ASSERT_TRUE(table1aMT == table1bMT);
+    ASSERT_FALSE(table1aMT == table2aMT);
+    ASSERT_FALSE(table1aMT.rawequal(table1bMT));
+    ASSERT_FALSE(table1aMT == table2b);
+    ASSERT_TRUE(table2aMT == table2aMT);
+    ASSERT_FALSE(table2aMT == table1bMT);
+
+#if LUABRIDGE_TEST_LUA_VERSION < 503
+    ASSERT_FALSE(table2aMT == table2b);
+#else
+    ASSERT_TRUE(table2aMT == table2b);
+#endif
```

Comment:
The reason for a different behavior isn't clear.
Looks like Lua 5.2 expected both tables have the same metatable, though the language reference says nothing about such check. It's basically states that a first found metamethod is used:
```
     function getbinhandler (op1, op2, event)
       return metatable(op1)[event] or metatable(op2)[event]
     end
```
So far I consider it's a bug in Lua 5.2.